soccer/Processor.cpp

```diff
@@ -696,23 +700,35 @@ void Processor::updateGeometryPacket(const SSL_GeometryFieldSize& fieldSize) {
     float adj = fieldSize.field_lines().Get(0).thickness() / 1000.0f / 2.0f;
 
     float fieldBorder = currentDimensions->Border();
+    // if (penaltyLongDist == 0 || penaltyShortDist == 0) {
+    //     cout << "CERO!" << endl;
+    // }
 
-    if (penalty != nullptr && center != nullptr && thickness != 0) {
+
+    if (penaltyLongDist != 0 && penaltyShortDist != 0 && center != nullptr && thickness != 0) {
         // Force a resize
         Field_Dimensions newDim = Field_Dimensions(
             fieldSize.field_length() / 1000.0f,
             fieldSize.field_width() / 1000.0f, fieldBorder, thickness,
             fieldSize.goal_width() / 1000.0f, fieldSize.goal_depth() / 1000.0f,
             Field_Dimensions::Default_Dimensions.GoalHeight(),
-            penalty->radius() / 1000.0f + adj,  // PenaltyDist
-            Field_Dimensions::Default_Dimensions.PenaltyDiam(),
-            penalty->radius() / 1000.0f + adj,       // ArcRadius
+            penaltyShortDist / 1000.0f,  // PenaltyS
+            penaltyLongDist / 1000.0f,
             center->radius() / 1000.0f + adj,        // CenterRadius
             (center->radius()) * 2 / 1000.0f + adj,  // CenterDiameter
             displacement / 1000.0f,                  // GoalFlat
             (fieldSize.field_length() / 1000.0f + (fieldBorder)*2),
             (fieldSize.field_width() / 1000.0f + (fieldBorder)*2));
 
+        if (newDim != *currentDimensions) {
+            // Set the changed field dimensions to the current ones
+            cout << "Updating field geometry based off of vision packet."
+                 << endl;
+            setFieldDimensions(newDim);
```

Comment:
Isn't this logic repeating what's a couple lines below it?

Reply:
Yes, is there some way you'd like me to write this?

Reply:
Hmm, I think that all we can do is move the cout inside the `setFieldDimensions` (and maybe the if statement check too, but that might make the call confusing). Do whatever you think looks better :)